elmslie-core/src/main/java/vivid/money/elmslie/core/store/binding/BindingStore.kt

```diff
@@ -0,0 +1,31 @@
+package vivid.money.elmslie.core.store.binding
+
+import vivid.money.elmslie.core.store.Store
+
+/**
+ * A store that binds two stores together and manages their lifecycle. In details:
+ * - [start] Starts both stores
+ * - [stop] Stops both stores
+ */
+internal class BindingStore<ParentEvent, ParentEffect, ParentState, ChildEvent, ChildEffect, ChildState>(
+    private val parent: Store<ParentEvent, ParentEffect, ParentState>,
+    private val child: Store<ChildEvent, ChildEffect, ChildState>,
+    contract: BindingContract<ParentEvent, ParentEffect, ParentState, ChildEvent, ChildEffect, ChildState>.() -> Unit
+) : Store<ParentEvent, ParentEffect, ParentState> by parent {
+
+    private val contracts = BindingContract(parent, child).apply(contract)
+
+    override fun start(): Store<ParentEvent, ParentEffect, ParentState> {
+        parent.start()
+        child.start()
+        contracts.apply()
+        return this
+    }
+
+    override fun stop() {
```

Comment:
Just an idea. Can it be helpful to save refs to contracts, that store involved to, inside stores to revoke contracts easily when a store is stopped?

Reply:
Contracts already have no effect when either of the stores stops and you suggest store code complication. No.